src/include/duckdb/parser/statement/insert_statement.hpp

```diff
@@ -31,6 +32,9 @@ class InsertStatement : public SQLStatement {
 	//! keep track of optional returningList if statement contains a RETURNING keyword
 	vector<unique_ptr<ParsedExpression>> returning_list;
 
+	//! CTEs
+	unordered_map<string, unique_ptr<CommonTableExpressionInfo>> cte_map;
```

Comment:
It seems like there is now a lot of duplicate logic around CTEs, e.g. the structure of this `cte_map`, the way it is added to the binder, the way it is copied, converted to a string, etc. I think it would be cleaner to extract this into a separate class (e.g. `CommonTableExpressionMap` or similar) and use that in each of the various statements.

Reply:
No need for inheritance here, we prefer not to use inheritance too much. I would just make a `CommonTableExpressionMap` class that holds all the methods associated with a set of CTEs.

Reply:
Sure, change is on the way